File: Examples/AspNetMvcCS/Controllers/DocumentViewerController.Overview.cs
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using GleamTech.AspNet;
using GleamTech.DocumentUltimate.AspNet;
using GleamTech.DocumentUltimate.AspNet.UI;
using GleamTech.Examples;

namespace GleamTech.DocumentUltimateExamples.AspNetMvcCS.Controllers
{
    public partial class DocumentViewerController
    {
        [AcceptVerbs(HttpVerbs.Get | HttpVerbs.Post)]
        public ActionResult Overview()
        {
            var exampleFileSelector = ViewBag.ExampleFileSelector = new ExampleFileSelector
            {
                Id = "exampleFileSelector",
                InitialFile = "Default.pdf",
                FormWrapped = false
            };

            var documentViewer = new DocumentViewer
            {
                Width = 800,
                Height = 600,
                Resizable = true,
                Document = exampleFileSelector.SelectedFile.ToString()
            };

            HandleSelectors(documentViewer);

            return View(documentViewer);
        }

        private void HandleSelectors(DocumentViewer documentViewer)
        {
            var context = Hosting.GetHttpContext();

            var selectedLanguage = context.Request["languageSelector"];
            if (selectedLanguage != null)
                documentViewer.DisplayLanguage = selectedLanguage;
            else
                selectedLanguage = documentViewer.DisplayLanguage;
            PopulateLanguageSelector(selectedLanguage);

            var selectedTheme = context.Request["themeSelector"];
            if (selectedTheme != null)
                documentViewer.Theme = selectedTheme;
            else
                selectedTheme = documentViewer.Theme;
            PopulateThemeSelector(selectedTheme);
        }

        private void PopulateLanguageSelector(string selectedLanguage)
        {
            ViewBag.LanguageList = new SelectList(
                DocumentUltimateWebConfiguration.AvailableDisplayCultures.Select(culture => new
                {
                    Value = culture.Name,
                    Text = culture.NativeName + $" ({culture.Name})"
                }),
                "Value",
                "Text",
                selectedLanguage
            );
        }

        private void PopulateThemeSelector(string selectedTheme)
        {
            ViewBag.ThemeList = new SelectList(
                new Dictionary<string, string>
                {
                    { "slate (Dark Mode: classic-dark)", "slate, classic-dark" },
                    { "classic-light (Dark Mode: classic-dark)", "classic-light, classic-dark" },
                    { "classic-dark", "classic-dark" }
                },
                "Value",
                "Key",
                selectedTheme
            );
        }
    }
}

File: Examples/AspNetMvcVB/Controllers/DocumentViewerController.Overview.vb

Imports System.Linq
Imports GleamTech.AspNet
Imports GleamTech.DocumentUltimate.AspNet
Imports GleamTech.DocumentUltimate.AspNet.UI
Imports GleamTech.Examples

Namespace Controllers
    Partial Public Class  DocumentViewerController
        Inherits Controller

        <AcceptVerbs(HttpVerbs.Get Or HttpVerbs.Post)>
	    Public Function Overview(fileSelector As String) As ActionResult
            Dim exampleFileSelector = New ExampleFileSelector() With {
                .Id = "exampleFileSelector",
                .InitialFile = "Default.pdf",
                .FormWrapped = False
            }
            ViewBag.ExampleFileSelector = exampleFileSelector

		    Dim documentViewer = New DocumentViewer() With { 
			    .Width = 800, 
			    .Height = 600,
                .Resizable = True,
			    .Document = exampleFileSelector.SelectedFile.ToString()
		    }

            HandleSelectors(documentViewer)

            Return View(documentViewer)
	    End Function

        Private Sub HandleSelectors(documentViewer As DocumentViewer)
            Dim context = Hosting.GetHttpContext()

            Dim selectedLanguage = context.Request("languageSelector")
            If selectedLanguage IsNot Nothing Then
                documentViewer.DisplayLanguage = selectedLanguage
            Else
                selectedLanguage = documentViewer.DisplayLanguage
            End If
            PopulateLanguageSelector(selectedLanguage)

            Dim selectedTheme = context.Request("themeSelector")
            If selectedTheme IsNot Nothing Then
                documentViewer.Theme = selectedTheme
            Else
                selectedTheme = documentViewer.Theme
            End If
            PopulateThemeSelector(selectedTheme)
        End Sub

        Private Sub PopulateLanguageSelector(selectedLanguage As String)
            ViewBag.LanguageList = New SelectList(
                DocumentUltimateWebConfiguration.AvailableDisplayCultures.Select(Function(culture) New With {
                    .Value = culture.Name,
                    .Text = culture.NativeName + $" ({culture.Name})"
                }),
                "Value",
                "Text",
                selectedLanguage
            )
        End Sub

        Private Sub PopulateThemeSelector(selectedTheme As String)
            ViewBag.ThemeList = New SelectList(
                New Dictionary(Of String, String)() From {
                    {"slate (Dark Mode: classic-dark)", "slate, classic-dark"},
                    {"classic-light (Dark Mode: classic-dark)", "classic-light, classic-dark"},
                    {"classic-dark", "classic-dark"}
                },
                "Value",
                "Key",
                selectedTheme)
        End Sub
    End Class
End Namespace
